Bottom: f24e419fa87e4b33c0ec18518cca505e9079f045
Top:    e66cc37e2617b7fd91400d9c10c8d3d335630c77
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-07-26 20:19:48 +0800

Refresh of tun-xdp-support

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 2b6696d..82bbce4 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1044,7 +1044,6 @@ static int tun_xdp_set(struct net_device *dev, struct bpf_prog *prog,
 static u32 tun_xdp_query(struct net_device *dev)
 {
 	struct tun_struct *tun = netdev_priv(dev);
-	struct tun_file *tfile = rtnl_dereference(tun->tfiles[0]);
 	const struct bpf_prog *xdp_prog;
 
 	xdp_prog = rtnl_dereference(tun->xdp_prog);
@@ -1507,6 +1506,8 @@ static ssize_t tun_get_user(struct tun_struct *tun, struct tun_file *tfile,
 		struct bpf_prog *xdp_prog;
 		int ret;
 
+		printk("generic!\n");
+
 		rcu_read_lock();
 		xdp_prog = rcu_dereference(tun->xdp_prog);
 		if (xdp_prog) {
